### ISSUE TITLE
Error while retrieving cache data

### ISSUE THREAD

**Issue by And96** [sur950/any_link_preview#48]
`I/flutter ( 3666): Error while retrieving cache data => type 'Null' is not a subtype of type 'String'`

Often appears this message in debug console

```
 any_link_preview.Metadata? metadata1 =
              await any_link_preview.AnyLinkPreview.getMetadata(
            link: link,
            cache: const Duration(days: 1),
          );
```

**Comment by AndreVicenteW** [comment]
Change to `cache: const Duration(days: 0)`

**Comment by And96** [comment]
cache dont work if i set Duration = 0

**Comment by MalayAgrawal** [comment]
`Error while retrieving cache data => type 'Null' is not a subtype of type 'String'`

Did you find any fix to it?

**Comment by And96** [comment]
> `Error while retrieving cache data => type 'Null' is not a subtype of type 'String'`
> 
> Did you find any fix to it?

No. There is no fix for this log message.

We must wait a fix from the plugin or create a fork.

**Comment by sur950** [comment]
Hi,

This is a long-standing issue and after a long debug, I found out that this is not happening on all devices and is happening only in a few devices. I am yet to get the pattern on what sort of device this is failing. Once I get that, I will find the issue going deeper into that OS/device and will fix this permanently. 

However, a quick fix is applied for this and will be out in 3.0.1.
